lib/cartopy/tests/mpl/test_mpl_integration.py

```diff
@@ -130,6 +130,37 @@ def test_global_scatter_wrap_no_transform():
     plt.scatter(x, y, c=data)
 
 
+@pytest.mark.natural_earth
+@ImageTesting(['global_hexbin_wrap'],
+              tolerance=12.4 if MPL_VERSION < '2.1.0' else 0.5)
+def test_global_hexbin_wrap_new_transform():
+    ax = plt.axes(projection=ccrs.PlateCarree())
+    # By default the coastline feature will be drawn after patches.
+    # By setting zorder we can ensure our hexbin polygons are drawn
+    # after the coastlines.
+    ax.coastlines(zorder=0)
+    x, y = np.meshgrid(np.linspace(-180, 180, 200), np.linspace(-90, 90, 200))
+    data = np.sin(np.sqrt(x**2 + y**2))
+    plt.hexbin(
+        x.flatten(),
+        y.flatten(),
+        C=data.flatten(),
+        transform=ccrs.PlateCarree(),
+        alpha=0.2,
+    )
+
+
+@pytest.mark.natural_earth
+@ImageTesting(['global_hexbin_wrap'],
```

Comment:
Probably only need one "good" image that you can compare multiple different methods against? Just make sure your expected data is always the same in each case and that the coordinates are what is changing being passed in.

Reply:
I am unsure how to specify the data that matches -180 to 180 in `PlateCarree`, the previous 0 to 360 would represent different points.

Reply:
Yeah, you will have to change your data calculation to make sure that the xs are mapped the same between the two cases. Something similar to this should work for those purposes I think.
```python
    x_wrap = np.where(x > 180, x-360, x)
    data = np.sqrt(x_wrap ** 2 + y ** 2)
```

Reply:
Eh, I can't make it work out to the same resulting image.  There are data differences and the coastlines are even slightly different :(

![result-global_hexbin_wrap-failed-diff](https://user-images.githubusercontent.com/210858/81063093-d488d100-8e9c-11ea-8618-e5ea68c0693a.png)

Reply:
Hm... That is odd. I wonder if the data limits aren't being updated properly so the extents of the images are slightly different? That also looks like the values for data weren't mapped in the transform properly either. I won't be able to take a closer look at this until this weekend or early next week.

Reply:
I think the issue is what is happening at the anti-meridian.

x range | transformed `xmin` | transformed `xmax`
--- | --- | ---
range(0, 360) | -179 | 180
range(-180, 180) | -180 | 179

Oh life is fun! `-180` appears to be the tough x value to get after transformation.  So I have pushed a slight jitter to the x values to prevent the anti-meridian from getting transformed.

:eyes: Travis-CI :)

Reply:
The transform must choose +180 when wrapping points, but if -180 is present doesn't transform it. Interesting, I didn't ever realize that. Your second case is leaving off the +180 point because range doesn't include the last value.

Rather than an offset of 0.1, I think you could use `arange(-179, 181)` and `arange(0, 360)` That way everything is just defined on the -179 -> 180 points in both cases...?